ecs/pod_definition_cache.go

```diff
@@ -0,0 +1,27 @@
+package ecs
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/evergreen-ci/cocoa"
+	"github.com/pkg/errors"
+)
+
+// PodDefinitionCache represents an external cache that tracks pod definitions.
+type PodDefinitionCache interface {
+	Put(ctx context.Context, res cocoa.ECSPodDefinitionOptions) error
+}
+
+// PodDefinitionManager manages pod definitions, which are configuration
+// templates used to run pods. It can be optionally backed by an external
+// PodDefinitionCache to keep track of the pod definitions.
+type PodDefinitionManager struct {
+	client cocoa.ECSClient    //nolint
+	cache  PodDefinitionCache //nolint
+}
+
+// CreatePodDefinition creates a pod definition and caches it.
+func (m *PodDefinitionManager) CreatePodDefinition(ctx context.Context, opts ...cocoa.ECSPodDefinitionOptions) (*ecs.TaskDefinition, error) {
```

Comment:
If the implementation will be done in separate tickets would it make sense to set up other functionalities you mentioned such as deleting?

Reply:
I'm going to add those later since I only thought through the creation step here. I want to spend some time thinking through deletion more before I write any more boilerplate.